content/blog/2023-12-18-torcx-deprecation-for-systemd-sysext.md

```diff
@@ -0,0 +1,82 @@
+  (Note that docker and containerd shipped with the base OS will continue to be updated as part of the base OS.)
+- Both containerd and docker can be disabled via ignition configs simply by overriding respective symlinks from `/etc/extensions/...`.
+  This can be useful when shipping a custom container runtime (or in cases where docker is simply not used).
+  When disabled, the respective binaries and libraries are not merged, and are missing from the OS filesystem.
+
+Starting with [Alpha-3794.0.0](https://www.flatcar.org/releases#release-3794.0.0), containerd and docker are included in the base OS image as sysexts.
+This means that in the regular Flatcar base OS image, there are no `containerd` or `docker` binaries in `/usr/bin` (where formerly the torcx wrapper scripts resided) - check out the [OS image contents](https://alpha.release.flatcar-linux.net/amd64-usr/current/flatcar_production_image_contents.txt) yourself!
+Instead, you'll find `containerd-flatcar.raw` and `docker-flatcar.raw` in `/usr/share/flatcar/sysext/`, and respective symlinks in `/usr/share/flatcar/etc/extensions/` (which is merged into `/etc/` at boot).
+
+
+**That's all well and good, but what does that mean for Flatcar users?**
+
+We aim for a seamless transition and have invested into significant test coverage to ensure user workloads are not disrupted by the change.
+Users will continue to enjoy a robust container OS experience, and do not need to do anything to adopt to the new system.
+
+As stated above, it is highly unlikely anyone outside the legacy OS builds even make use of torcx (or heard of it, for that matter).
+Just to be on the safe side, we will continue to loudly announce the move to sysext throughout the next releases, i.e. when the torcx removal will go Beta, and later graduate to Stable.
+
+Should you, against all probability, indeed make use of torcx customisations in your deployments, then we'd like to hear from you!
+It should be really straightforward migrating the customisations to sysext, and chances are high that adapting sysext will improve your maintenance and operations experience just as it did for us.
+Please reach out to us via our [issue tracker](https://github.com/flatcar/Flatcar/issues/new/choose), [Matrix](https://app.element.io/#/room/#flatcar:matrix.org) or [Slack](https://kubernetes.slack.com/archives/C03GQ8B5XNJ) chat, or join our [office hours](https://github.com/flatcar/Flatcar/discussions/categories/flatcar-office-hours) for a face-to-face chat!
+
+**Docker 24 incoming**
+
+On a closing note, moving from custom torcx docker / containerd ebuilds to upstream Gentoo ones significantly eases the maintenance burden and makes integration of major version updates significantly easier.
+So the new Alpha versions also ship docker 24, which is a major version bump from version 20 currently in Beta and Stable, as well as the latest containerd.
```

Comment:
if we're shipping the docker sysext, should we also ship podman? there's some demand for it.

Reply:
Also, WASM. We're already tracking this, will add a section on that and some references.